### PR TITLE
doc/1: Add requirements and design

### LINKED ISSUE #1: Add requirements and design
document requirements and design for the document virtual schema project.

### PR DESCRIPTION
**Pull request by jakobbraun**
Closes #1

### CODE REVIEW

File: features.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+The relational model of Exasol does not support nested lists.
+
+###  Filter Push-Down
+
+`feat~filer-pushdown~1`
+
+The Virtual Schema adapter pushes as many predicates from the WHERE clause as possible down to the data source.
+
+Needs: generic design, design in each variant
```
Comment (tkilias) [inline features.md]: Not sure, if OpenFastTrace takes this. Alternatively, you can define new artifact types, such as gdsn, vdsn or ddsn. OFT doesn't restrict you regarding this.

File: requirements.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+Keep the required memory footprint for the UDFs low.
+
+Rationale:
+
+By default, an Exasol cluster node reserves ~ 4 GB RAM for UDFs.
+It's the minimum of the DB-RAM (2 GB on a default docker-db, much higher in clusters), a quota for RAM for all processes (~ 32 GB per node) on a node and a quote for the RAM for one SQL session (~ 4 GB per node).
+That's independent of how much RAM is available on that node and on how many CPUs it has.
+That RAM quota is shared by all UDF instances in a query.
+Onn the other hand, for a quick loading parallelization is usually crucial.
```
Comment (tkilias) [inline requirements.md]:
```suggestion
On the other hand, for a quick loading parallelization is usually crucial.
```

File: requirements.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+### Filter Pushdown
+
+`req~filter-pushdown~1`
+
+Each datasource implements pushing down predicates to the data source.
+
+Rationale:
+
+Each datasource can have a different API. There is no generic methos for sending the predicates.
```
Comment (tkilias) [inline requirements.md]:
```suggestion
Each datasource can have a different API. There is no generic method for sending the predicates.
```

File: requirements.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+Covers:
+
+* `feat~filer-pushdown~1`
+
+Needs: dsn per dialect
```
Comment (tkilias) [inline requirements.md]: ddsn or vdsn, as suggested

File: requirements.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+### Query Splitting
+
+`req~query-splitting~1`
+
+The Virtual Schema adapter can transform the WHERE clause of a query into a push-down-selection and a post-selection. The push-down selection only contains predicates that can be pushed down to the data source. The post selection is applied after the transfer in the Exasol DB.
+
+The result mus always be exactly the same as if we load the whole table and apply the WHERE clause in the Exasol DB.
```
Comment (tkilias) [inline requirements.md]:
```suggestion
The result must always be exactly the same as if we load the whole table and apply the WHERE clause in the Exasol DB.
```

File: requirements.md [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+### Source Reference Column
+
+`req~source-reference-column~1`
+
+This virtual schema offers a `SOURCE_REFERENCE` column that contains the name of the file or table the data originates from.
+
+Rationale:
+
+Sometimes that also the filename can contain important information. For example the date of a log file. That's of special relevance when selecting the source via a wildcard and by that the table in Exasol contains data from multiple files.
```
Comment (tkilias) [inline requirements.md]:
```suggestion
Sometimes also the filename can contain important information. For example the date of a log file. That's of special relevance when selecting the source via a wildcard and by that the table in Exasol contains data from multiple files.
```

**Review by tkilias — CHANGES_REQUESTED** [review]

File: features.md [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+* DynamoDB <!-- `feat~dynamodb~1` -->
+* Files
+    * with file types:
+        * JSON
+        * JSON-lines
+        * Parquet
+    * with storage backends
+        * from AWS S3
+        * from Google Cloud Storage
+        * from BucketFS
```
Comment (tkilias) [inline features.md:48]: do we need separate features for this?

File: features.md [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+### Fast Loading for Different Load Patterns
+
+The Virtual Schema performs well for the following load patterns:
+
+* Many small files
+* One / few big files
+* One / few small files
```
Comment (tkilias) [inline features.md:58]: do we need separate requirements for this?

File: features.md [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+* Files
+    * with file types:
+        * JSON
+        * JSON-lines
+        * Parquet
+    * with storage backends
+        * from AWS S3
+        * from Google Cloud Storage
+        * from BucketFS
+
```
Comment (pj-spoelders) [inline features.md]:
```suggestion
        * from Azure Blob Storage
        * from Azure Data Lake Gen2 Storage
        
```

**Review by pj-spoelders — DISMISSED** [review]

File: features.md (hunk 7 above)
Comment (jakobbraun) [reply features.md:48]: In the future probably yes. Right now I just published it without OFT tracing. There I prefer the compact notation.

File: features.md (hunk 8 above)
Comment (jakobbraun) [reply features.md:58]: same

**Review by pj-spoelders — APPROVED** [review]

File: features.md (hunk 7 above)
Comment (tkilias) [reply features.md:48]: Ok sounds good. Maybe we should create a ticket for that change, such that we don't forget it.

Comment (jakobbraun) [reply features.md:48]: done https://github.com/exasol/document-virtual-schema-spec/issues/3

**Review by tkilias — APPROVED** [review]

**Pull request merged**